EIPS/eip-3525.md

```diff
@@ -0,0 +1,422 @@
+        },
+         "decimals": {
+            "type": "integer",
+            "description": "The number of decimal places that the token amount should display - e.g. 18, means to divide the token amount by 1000000000000000000 to get its user representation."
+        },
+        "properties": {
+            "type": "object",
+            "description": "Arbitrary properties. Values may be strings, numbers, object or arrays."
+        }
+    }
+}
+```
+
+### Approval
+
+VNFT adds a new approval model, that is, one can approve operators to partial transfer units from a token with certain ID, the new interface is:
+```function approve(address to, uint256 tokenId, uint256 units); ```
+
+
+
+## Rationale 
+
+### Metadata generation
+
+Since VNFT is designed for representing underlying assets, rather than artifacts for gaming or arts, the implementation should give out the metadata directly from contract code, rather than give a URL of a server for returning metadata.
+
+
+### Design decision: Keep unsafe transfer
+
+There are mainly two reasons we keep the unsafe transfer interfaces:
+
+1. Since VNFT is ERC721 compatible, we must keep compatibility for all wallets and contracts that are still calling unsafe transfer interfaces for ERC721 tokens.
+2. We want to keep the ability that dapps can trigger business logic on contracts by simply transferring VNFT tokens to them, that is, a contract can put business logic in `onVNFTReceived` function so that it can be called whenever a token is transferred using `safeTransferFrom`. However, in this situation, an approved contract with customized transfer functions like deposit etc. SHOULD never call `safeTransferFrom` since it will result in confusion that whether `onVNFTReceived` is called by itself or other dapps that safe transfer a token to it.
+
+
+### Approval
+
+For maximum semantical compatibility with ERC721, as well as simplifying the approval model, we decided to make the relationship between two levels of approval like that:
+
+1. Approval of an id does not result in the ability to partial transfer units from this id by the approved operator;
+2. Approval of all units in a token does not result in the ability to transfer the token entity by the approved operator;
+3. `setApprovalForAll` will result in the ability to transfer any tokens from the owner, as well as the ability to partial transfer units from any token.
+3. `setApprovalForAll` will result in the ability to approve any tokens of the owner to third parties, as well as the ability to approve partial transfer units of any token to third parties.
+
+
+## Backwards Compatibility
+	
+As mentioned at the very beginning, a VNFT contract is basically an ERC721 contract, hence it is 100% compatible with ERC721.
+
+
+## References
```

Comment:
This isn't a valid top level section of an ERC per [EIP-1](https://eips.ethereum.org/EIPS/eip-1), please remove this section.

Reply:
Still waiting on this too.

Reply:
Hi, thanks for the reminding, we had correct the names/emails now, but do you mean References is not a top level section of ERC? I saw from EIP-1 that there is a References section, not sure wether there is some misunderstanding on this, thanks in advance.

Reply:
@will-edge I think you are confusing "Reference" and "Reference Implementation". There is no "Reference" section in EIPs.